diff --git a/core/libqutim.qbs b/core/libqutim.qbs
index 1d833a4..9a9f6ff 100755
--- a/core/libqutim.qbs
+++ b/core/libqutim.qbs
@@ -50,21 +50,19 @@ Framework {
     cpp.objcxxFlags: {
         var flags = base.concat("-std=c++11");
         if (qbs.toolchain.contains("clang"))
-            flags = flags.concat("-stdlib=libc++");
+            flags = flags.concat("-stdlib=libstdc++");
         return flags;
     }
     cpp.cxxFlags: {
         var flags = base.concat("-std=c++11");
         if (qbs.toolchain.contains("clang"))
-            flags = flags.concat("-stdlib=libc++");
+            flags = flags.concat("-stdlib=libstdc++");
         return flags;
     }
     cpp.linkerFlags: {
         var flags = base;
         if (qbs.toolchain.contains("clang"))
-            flags = flags.concat(["-stdlib=libc++"])
-        if (qbs.toolchain.contains("clang") && qbs.targetOS.contains("linux"))
-            flags = flags.concat("-lcxxrt");
+            flags = flags.concat(["-stdlib=libstdc++"])
         return flags;
     }
     cpp.minimumOsxVersion: "10.8"
@@ -87,21 +85,19 @@ Framework {
         cpp.cxxFlags: {
             var flags = base.concat("-std=c++11");
             if (qbs.toolchain.contains("clang"))
-                flags = flags.concat("-stdlib=libc++");
+                flags = flags.concat("-stdlib=libstdc++");
             return flags;
         }
         cpp.objcxxFlags: {
             var flags = base.concat("-std=c++11");
             if (qbs.toolchain.contains("clang"))
-                flags = flags.concat("-stdlib=libc++");
+                flags = flags.concat("-stdlib=libstdc++");
             return flags;
         }
         cpp.linkerFlags: {
             var flags = base;
             if (qbs.toolchain.contains("clang"))
-                flags = flags.concat(["-stdlib=libc++"])
-            if (qbs.toolchain.contains("clang") && qbs.targetOS.contains("linux"))
-                flags = flags.concat("-lcxxrt");
+                flags = flags.concat(["-stdlib=libstdc++"])
             return flags;
         }
         cpp.minimumOsxVersion: "10.8"
diff --git a/protocols/vkontakte/vreen/modules/vreen/core/core.qbs b/protocols/vkontakte/vreen/modules/vreen/core/core.qbs
index 0388a63..2ba5e63 100644
--- a/protocols/vkontakte/vreen/modules/vreen/core/core.qbs
+++ b/protocols/vkontakte/vreen/modules/vreen/core/core.qbs
@@ -22,7 +22,7 @@ Module {
     cpp.cxxFlags: {
         var flags = base.concat("-std=c++11");
         if (qbs.toolchain.contains("clang"))
-            flags = flags.concat("-stdlib=libc++");
+            flags = flags.concat("-stdlib=libstdc++");
         return flags;
     }
