diff --git a/./gogoc-tsp/src/tsp/tsp_setup.c-orig b/./gogoc-tsp/src/tsp/tsp_setup.c
index 51b6a62..c7361d8 100644
--- a/./gogoc-tsp/src/tsp/tsp_setup.c-orig
+++ b/./gogoc-tsp/src/tsp/tsp_setup.c
@@ -185,8 +185,11 @@ void set_tsp_env_variables( const tConf* pConfig, const tTunnel* pTunnelInfo )
   gTunnelInfo.szIPV6AddrLocalEndpoint = pTunnelInfo->client_address_ipv6;
 
   // Specify client dns IPv6 address
-  tspSetEnv("TSP_CLIENT_DNS_ADDRESS_IPV6", pTunnelInfo->client_dns_server_address_ipv6, 1);
-  gTunnelInfo.szIPV6AddrDns = pTunnelInfo->client_dns_server_address_ipv6;
+  if( pTunnelInfo->client_dns_server_address_ipv6 != NULL)
+  {
+    tspSetEnv("TSP_CLIENT_DNS_ADDRESS_IPV6", pTunnelInfo->client_dns_server_address_ipv6, 1);
+    gTunnelInfo.szIPV6AddrDns = pTunnelInfo->client_dns_server_address_ipv6;
+  }
 
   // Specify local endpoint domain name
   if( pTunnelInfo->client_dns_name != NULL)
