--- Crypto/Aes_hw_cpu.asm.old	2016-10-17 19:40:41.000000000 +0300
+++ Crypto/Aes_hw_cpu.asm	2016-10-20 14:06:22.980365161 +0300
@@ -331,3 +331,12 @@
 
 
 %endif	; __BITS__ != 16
+%ifidn __OUTPUT_FORMAT__,elf
+	section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+	section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+	section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
--- Crypto/AesSmall_x86.asm.old	2016-10-17 19:40:41.000000000 +0300
+++ Crypto/AesSmall_x86.asm	2016-10-20 14:06:48.562320775 +0300
@@ -1442,3 +1442,12 @@
     db  v8(0xe1),v8(0x69),v8(0x14),v8(0x63),v8(0x55),v8(0x21),v8(0x0c),v8(0x7d)
 
 %endif
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
--- Crypto/Aes_x64.asm.old	2016-10-17 19:40:41.000000000 +0300
+++ Crypto/Aes_x64.asm	2016-10-20 14:07:37.549771358 +0300
@@ -905,3 +905,12 @@
 %endif
 
 %endif
+%ifidn __OUTPUT_FORMAT__,elf
+	section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+	section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+	section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
--- Crypto/Aes_x86.asm.old	2016-10-17 19:40:41.000000000 +0300
+++ Crypto/Aes_x86.asm	2016-10-20 14:09:20.828482064 +0300
@@ -644,3 +644,12 @@
     do_exit
 
 %endif
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
