*** Makefile.old	2014-03-26 06:21:29.648994033 +0400
--- Makefile	2014-03-26 06:42:09.074081942 +0400
***************
*** 25,28 ****
  	install -c -m 0644 ifd-ASEDriveIIIe-USB.bundle/Contents/Resources/English.lproj/InfoPlist.strings "${DRIVER_DIR}/Contents/Resources/English.lproj"
  	install -c -d "${DRIVER_DIR}/Contents/Linux"
  	install -c -m 0755 libASEDriveIIIe-USB.so "${DRIVER_DIR}/Contents/Linux"
- 	install -c -m 0644 50-pcscd-asedriveiiie.rules /etc/udev/rules.d/.
--- 25,27 ----
